utilities/include/vec4.hpp

```diff
@@ -0,0 +1,171 @@
+#ifndef VEC4_HPP
+#define VEC4_HPP
+
+#include <cstddef>
+#include <ostream>
+#include <cmath>
+#include <xsimd/xsimd.hpp>
```

Comment:
It would be nice if you put the `xsimd` include after the STL includes, with a blank line in between (similar to what you did for `config.hpp`.